src/sharing/send.js

```diff
@@ -54,9 +55,9 @@ export const handleSend = async (data: SendStream | SendPm, auth: Auth, _: GetTe
 
   try {
     await sendMessage(auth, messageData);
+    showToast(_('Message sent'));
   } catch (err) {
     showToast(_('Failed to send message'));
-    return;
+    logging.error(err);
   }
-  showToast(_('Message sent'));
```

Comment:
This commit message says:

    sharing: Capture error via sentry if sending shared data fails.

and that change sounds great. The commit is making a couple of other changes too, though, which makes it a bit confusing to follow.

How about just adding the `logging.error` line before the existing `return`?

I also like keeping the `try` tightly scoped around the `await`, because that means the error handling in the `catch` block gets to be tightly focused and know exactly where the error was. For example if there's an error somehow in the `showToast` with the "sent" text, then if that's inside the `try` block we'd end up trying to handle it by saying "Failed to send message", which wouldn't be right.

Reply:
For cross-reference: I've expanded this to an entry in our style guide: d99ad3ad2